packages/use-query-params/src/__tests__/index.tsx

```diff
@@ -1,30 +1,41 @@
 import { act, renderHook } from '@testing-library/react-hooks'
 import { History, createMemoryHistory } from 'history'
-import React, { ReactNode } from 'react'
+import React, { ReactNode, useLayoutEffect, useState } from 'react'
 import { MemoryRouter, Router } from 'react-router-dom'
 import useQueryParams from '..'
 
 const wrapper =
-  ({ pathname = 'one', search, history }: { pathname?: string, search: string, history?: History }) =>
-  // eslint-disable-next-line react/prop-types
-  ({ children }: { children: ReactNode }) =>
-    (
-      history ? (
-        <Router history={history}>
+  ({ pathname = '/one', search, history }: { pathname?: string, search: string, history?: History }) =>
+  ({ children }: { children: ReactNode }) => {
+
+    if (history) {
+      const [state, setState] = useState({
+        action: history.action,
+        location: history.location
+      });
+
+      useLayoutEffect(() => history.listen(setState), []);
```

Comment:
Shouldn't hooks be nested in `if` statements?

Reply:
Only if it affects render, this state is not used outside of the if scope, there should be no issues :)